sdk/core/System.ClientModel/src/Convenience/ActivityExtensions.cs

```diff
@@ -0,0 +1,121 @@
+                                                ActivityKind kind = ActivityKind.Internal,
+                                                ActivityContext parentContext = default,
+                                                IEnumerable<KeyValuePair<string, object?>>? tags = null)
+    {
+        if (options.EnableDistributedTracing == false)
+        {
+            return null;
+        }
+
+        bool shouldSuppressNested = kind == ActivityKind.Client || kind == ActivityKind.Internal;
+        bool isInnerSpan = ScmScopeValue.Equals(Activity.Current?.GetCustomProperty(ScmScopeLabel));
+        if (shouldSuppressNested && isInnerSpan)
+        {
+            return null;
+        }
+
+        Activity? activity = activitySource.StartActivity(name, kind, parentContext, tags);
+
+        if (shouldSuppressNested)
+        {
+            activity?.SetCustomProperty(ScmScopeLabel, ScmScopeValue);
+        }
+
+        return activity;
+    }
+
+    /// <summary>
+    /// Marks the <paramref name="activity"/> as failed.
+    /// </summary>
+    /// <remarks>This method should only be used by client library authors and should not
+    /// be used by consuming applications. Applications should use the System.Diagnostics package
+    /// to implement distributed tracing.</remarks>
+    /// <param name="activity">The activity to mark as failed.</param>
+    /// <param name="exception">The <see cref="Exception"/> encountered during the operation.</param>
+    public static Activity MarkFailed(this Activity activity, Exception? exception)
+    {
+        // See: https://opentelemetry.io/docs/specs/semconv/general/recording-errors/
+
+        // SHOULD set the span status code to error
+        // When the operation fails with an exception the span status description SHOULD be set to the
+        // exception message
+        activity.SetStatus(ActivityStatusCode.Error, exception?.Message);
+
+        // SHOULD set error.type see https://opentelemetry.io/docs/specs/semconv/attributes-registry/error/#error-type
+        // This instrumentation uses [HTTP status code] or [full name of the exception type] or [_OTHER] in that order
+        string? errorCode = null;
+        if (exception is ClientResultException clientResultException)
+        {
+            errorCode = clientResultException.Status.ToString();
+        }
+        errorCode ??= exception?.GetType().FullName;
+        errorCode ??= "_OTHER";
+
+        activity.SetTag("error.type", errorCode);
+
+        // If there is an exception, SHOULD record this exception as a span event
+        // see https://opentelemetry.io/docs/specs/semconv/general/recording-errors/#recording-exceptions
+        if (exception == null)
+        {
+            return activity;
+        }
+        // TODO - we should switch to the runtime AddException added in .NET 9 when we can
+        // see https://github.com/Azure/azure-sdk-for-net/issues/49121
+        activity.AddException(exception);
```

Comment:
let's not call it. There is an endless debate in OTel around it and chances are the underlying OTel thing (span events) is going to be deprecated soon - https://github.com/open-telemetry/opentelemetry-specification/pull/4430

Exception stack traces are huge and verbose and people complain about us reporting them all the time.

We'll also log this exception, so if someone needs it, they can still get this information